azurerm/internal/services/mariadb/data_source_mariadb_server.go

```diff
@@ -0,0 +1,149 @@
+					regexp.MustCompile("^[-a-zA-Z0-9]{3,50}$"),
+					"MariaDB server name must be 3 - 50 characters long, contain only letters, numbers and hyphens.",
+				),
+			},
+
+			"location": azure.SchemaLocationForDataSource(),
+
+			"resource_group_name": azure.SchemaResourceGroupNameForDataSource(),
+
+			"sku_name": {
+				Type:     schema.TypeString,
+				Computed: true, // remove in 2.0
+			},
+
+			"administrator_login": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"administrator_login_password": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+
+			"version": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"storage_profile": {
+				Type:     schema.TypeList,
+				Computed: true,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"storage_mb": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"backup_retention_days": {
+							Type:     schema.TypeInt,
+							Computed: true,
+						},
+
+						"geo_redundant_backup": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+
+						"auto_grow": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+					},
+				},
+			},
+
+			"ssl_enforcement": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"fqdn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+
+			"tags": tags.SchemaDataSource(),
+		},
+	}
+}
+
+func dataSourceMariaDbServerRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*clients.Client).MariaDB.ServersClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	name := d.Get("name").(string)
+	resourceGroup := d.Get("resource_group_name").(string)
+
+	resp, err := client.Get(ctx, resourceGroup, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			return fmt.Errorf("Error: Azure MariaDB Server %q (Resource Group %q) was not found", name, resourceGroup)
+		}
+
+		return fmt.Errorf("Error making Read request on Azure MariaDB Server %s: %+v", name, err)
+	}
+
+	d.Set("name", resp.Name)
+	d.Set("resource_group_name", resourceGroup)
+
+	if location := resp.Location; location != nil {
+		d.Set("location", azure.NormalizeLocation(*location))
+	}
+
+	if sku := resp.Sku; sku != nil {
+		d.Set("sku_name", sku.Name)
+	}
+
+	if properties := resp.ServerProperties; properties != nil {
+		d.Set("administrator_login", properties.AdministratorLogin)
+		d.Set("version", string(properties.Version))
+		d.Set("ssl_enforcement", string(properties.SslEnforcement))
+		// Computed
+		d.Set("fqdn", properties.FullyQualifiedDomainName)
+
+		if err := d.Set("storage_profile", flattenMariaDbStorageProfile(properties.StorageProfile)); err != nil {
+			return fmt.Errorf("Error setting `storage_profile`: %+v", err)
+		}
+	}
+
+	if err := d.Set("sku", flattenMariaDbServerSku(resp.Sku)); err != nil {
```

Comment:
This doesn't exist in schema?

Reply:
I looked right over that. Thanks for catching it!